src/classes/smith/smith.ts

```diff
@@ -0,0 +1,123 @@
+import _ from "lodash";
+import { IBehaviour } from "../../types/character/behaviour";
+import { IEquipment } from "../../types/shop/equipment";
+import { IItem } from "../../types/shop/item";
+import { ISmith } from "../../types/character/smith";
+import { IStat } from "../../types/character/stat";
+import { IWeapon } from "../../types/shop/weapon";
+import { SmithTypesEnum } from "../../enums/smith-types.enum";
+import { EquipmentSpotsEnum } from "../../enums/equipment-spots.enum";
+import { ObjectTypesEnum } from "../../enums/object-types.enum";
+import { StatNamesEnum } from "../../enums/stat-names.enum";
+import { SmithBehaviourEnum } from "../../enums/smith-behaviour";
+import { WeaponCategoriesEnum } from "../../enums/weapon-categories.enum";
+import Smiths from "../../data/models/smith-schema";
+import names from "../../data/texts/smiths/names.json";
+
+export class Smith implements ISmith {
+  public behaviour: IBehaviour;
+  public description: string;
+  public equipment?: IEquipment[];
+  public name: string;
+  public items?: IItem[];
+  public startingLevel: number;
+  public stats: IStat[];
+  public type: SmithTypesEnum;
+  public weapons?: IWeapon[];
+
+  constructor() {
+    this.behaviour = this._generateBehaviour();
+    this.description = ``;
+    this.name = this._generateName();
+    this.startingLevel = this._generateStartingLevel(10);
+    this.stats = this._generateStats();
+    this.type = this._generateType();
+    this.weapons = this._generateWeapons(_.random(1, 3));
+  }
+
+  public async generateEntity(): Promise<ISmith> {
+    const smith: ISmith = {
+      behaviour: this.behaviour,
+      description: this.description,
+      name: this.name,
+      startingLevel: this.startingLevel,
+      stats: this.stats,
+      type: this.type,
+      weapons: this.weapons,
+    };
+
+    void Smiths.create(smith);
+
+    return Promise.resolve(smith);
+  }
+
+  private _generateBehaviour(): IBehaviour {
+    const behaviour: IBehaviour = {
+      description: ``,
+      name: _.sample(SmithBehaviourEnum) || SmithBehaviourEnum.SHY,
+    };
+
+    return behaviour;
+  }
+
+  private _generateName(): string {
+    const name = _.sample(names.names);
+
+    if (name) {
+      return name;
+    }
+    return `Bob`;
+  }
+
+  private _generateStartingLevel(max: number): number {
+    return _.random(1, max);
+  }
+
+  private _generateStats(): IStat[] {
+    const generatedStats: IStat[] = [];
+
+    _.forEach(StatNamesEnum, (statName: StatNamesEnum): void => {
```

Comment:
I think using a _.reduce would improve slightly your code quality.

Reply:
Ok. I will look at how to use it ^^